java/arcs/core/analysis/DependencyNode.kt

```diff
@@ -84,56 +84,150 @@ private typealias Path = List<Identifier>
  */
 sealed class DependencyNode {
 
+  /** Path of [Identifier]s representing access to a handle connection. */
+  abstract val path: Path
+
+  /** Association with the current node and child nodes, via the children's Identifier. */
+  abstract val associations: Map<Identifier, DependencyNode>
+
+  /** Set of [DependencyNode]s that the current node depends on. */
+  abstract val dependency: Set<DependencyNode>
```

Comment:
In the current state of the PR, this goes unused. There's mixed levels for representations of dependency edges.

Reply:
I thought you showed me a test example where the dependency edges was explicitly present?